cli/src/pcluster/models/imagebuilder.py

```diff
@@ -187,6 +213,24 @@ def create(self, disable_rollback: bool = False):
             LOGGER.critical(e)
             raise ImageBuilderActionError(f"ImageBuilder stack creation failed.\n{e}")
 
+    def delete(self):
+        """Delete CFN Stack and associate resources and deregister the image."""
+        # validate image name
+        self._validate_image_name()
```

Comment:
not clear to me why we need to validate the image name at deletion time.
If it's an important step we should do it in the api or in the creation of the image object.

We might add validators for the syntax of the parameters ar first step of the api.

Reply:
validate_image_name idea here is for the different regex patterns between image name and cfn stack name.
Although it is a corner case, for example, if a name is suitable for cfn stack name, without validation, we will pass this name to generate cdk template and create cfn stack. We will meet error during cfn stack creation due to invalid image name. But if we validate before cfn stack creation, we will avoid that failure.